apps/files_trashbin/lib/Trashbin.php

```diff
@@ -189,7 +189,10 @@ private static function copyFilesToUser($sourcePath, $owner, $targetPath, $user,
 
 		$target = $user . '/files_trashbin/files/' . $targetFilename . '.d' . $timestamp;
 		$source = $owner . '/files_trashbin/files/' . $sourceFilename . '.d' . $timestamp;
-		self::copy_recursive($source, $target, $view);
+		$free = $view->free_space($target);
+		if (($free < 0) || ($view->filesize($source) < $free)) {
```

Comment:
Then I would propose a more readable `if` statement:

```suggestion
		$isUnknownOrUnlimitedFreeSpace = $free < 0;
		$isEnoughFreeSpaceLeft = $view->filesize($source) < $free;
		if ($isUnknownOrUnlimitedFreeSpace || $isEnoughFreeSpaceLeft) {
```